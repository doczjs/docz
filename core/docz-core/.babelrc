{
  "plugins": ["lodash"],
  "presets": [
    "@babel/preset-typescript",
    [
      "@babel/preset-env",
      {
        "targets": {
          "node": "current"
        }
      }
    ]
  ]
}
