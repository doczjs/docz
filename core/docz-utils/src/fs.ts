import * as path from 'path'
import * as fs from 'fs-extra'
import { compile } from 'art-template'

import { format } from './format'

export const touch = (file: string, raw: string, formatFiles: boolean) =>
  new Promise(async (resolve, reject) => {
    const content = /jsx?$/.test(path.extname(file))
      ? await format(raw, formatFiles)
      : raw
    const stream = fs.createWriteStream(file)

    stream.write(content, 'utf-8')
    stream.on('finish', () => resolve())
    stream.on('error', err => reject(err))
    stream.end()
  })

export const compiled = (file: string, opts: Record<string, any> = {}) => {
  const raw = fs.readFileSync(file, { encoding: 'utf-8' })
  return compile(raw, opts)
}
