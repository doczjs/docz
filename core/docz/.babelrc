{
  "plugins": ["@babel/plugin-syntax-dynamic-import", "lodash"]
}
