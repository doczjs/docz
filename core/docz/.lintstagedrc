{
  "*.{ts,tsx}": ["npm run fix", "git add"]
}
