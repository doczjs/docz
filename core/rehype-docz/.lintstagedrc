{
  "*.{ts,tsx}": ["yarn fix", "git add"]
}
