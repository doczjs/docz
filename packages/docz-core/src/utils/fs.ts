import * as path from 'path'
import * as fs from 'fs-extra'
import { compile } from 'art-template'
import { format } from 'docz-utils'

export const touch = (file: string, raw: string) =>
  new Promise(async (resolve, reject) => {
    const content = /jsx?$/.test(path.extname(file)) ? await format(raw) : raw
    const stream = fs.createWriteStream(file)

    stream.write(content, 'utf-8')
    stream.on('finish', () => resolve())
    stream.on('error', err => reject(err))
    stream.end()
  })

export const read = (file: string): Promise<string> =>
  new Promise<string>((resolve, reject) => {
    let data = ''
    const stream = fs.createReadStream(file, { encoding: 'utf-8' })

    stream.on('data', chunk => (data += chunk))
    stream.on('end', () => resolve(data))
    stream.on('error', err => reject(err))
  })

export const readIfExist = async (file: string): Promise<string | null> => {
  const exist = await fs.pathExists(file)
  return exist ? read(file) : Promise.resolve(null)
}

export const compiled = async (
  file: string,
  opts: Record<string, any> = {}
): Promise<(args: any) => string> =>
  read(file)
    .then(data => compile(data, opts))
    .catch(err => err)
