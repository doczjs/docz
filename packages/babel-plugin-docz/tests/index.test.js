import { transform } from '@babel/core';
import fs from 'fs';
import path from 'path';
import plugin from '../src';

let fixture = path.resolve('./tests/fixtures/example.js');

const code = fs.readFileSync(fixture);

it('works', () => {
  const { code: result } = transform(code, {
    plugins: [plugin],
    filename: fixture
  });

  expect(result).toMatchSnapshot();
});
