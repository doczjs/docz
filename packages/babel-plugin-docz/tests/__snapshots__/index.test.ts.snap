// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
"\\"use strict\\";

/**
 * Some description
 */
function get(object, path) {
  return object[path];
}
/**
 * Some description
 */


get.__docz = {
  name: \\"get\\",
  filename: \\"/Users/marcelopiva/workspace/docz/packages/babel-plugin-docz/tests/fixtures/example.js\\"
};

class Abc {
  /**
   * Some description
   */
  method() {
    return null;
  }

}

Abc.__docz = {
  name: \\"Abc\\",
  filename: \\"/Users/marcelopiva/workspace/docz/packages/babel-plugin-docz/tests/fixtures/example.js\\"
};"
`;
